Remove C10_HOST_CONSTEXPR_EXCEPT_WIN_CUDA and CONSTEXPR_EXCEPT_WIN_CUDA

r-barnes opened the pull request:
BC linter suppressed due to removal of `tools/linter/adapters/constexpr_linter.py`

malfet reviewed (approved):
Please keep defines in place (but hide them behind the same deprecated logic as you did for C10_DEPRECATED)

malfet commented:
@r-barnes why did you added suppress BC linter? Would be good to add this to PR description, if you don't believe those signals are accurate

r-barnes committed:
Remove C10_HOST_CONSTEXPR_EXCEPT_WIN_CUDA and CONSTEXPR_EXCEPT_WIN_CUDA (#138479)

Summary:
Fixes #ISSUE_NUMBER

Pull Request resolved: https://github.com/pytorch/pytorch/pull/138479

Test Plan: Sandcastle

Reviewed By: malfet

Differential Revision: D64785175

Pulled By: r-barnes
